wpilibc/src/main/native/cpp/DriverStation.cpp

```diff
@@ -26,41 +28,70 @@
 #include "frc/WPIErrors.h"
 
 namespace frc {
+// A simple class which caches the previous value written to an NT entry
+// Used to prevent redundant, repeated writes of the same value
+template <class T>
+class MatchDataSenderEntry {
+ public:
+  MatchDataSenderEntry(std::shared_ptr<nt::NetworkTable> table,
+                       const std::string& key, const T& initialVal) {
+    static_assert(std::is_convertible<decltype(initialVal), bool>() ||
+                      std::is_convertible<decltype(initialVal), double>() ||
+                      std::is_convertible<decltype(initialVal), std::string>(),
+                  "Invalid type for MatchDataSenderEntry - must be convertable "
+                  "to bool, double or std::string");
+
+    ntEntry = table->GetEntry(key);
+    if constexpr (std::is_convertible<decltype(initialVal), bool>()) {
+      ntEntry.ForceSetBoolean(initialVal);
+    } else if constexpr (std::is_convertible<decltype(initialVal), double>()) {
+      ntEntry.ForceSetDouble(initialVal);
+    } else if constexpr (std::is_convertible<decltype(initialVal),
+                                             std::string>()) {
+      ntEntry.ForceSetString(initialVal);
+    }
+    prevVal = initialVal;
+  }
+  void Set(const T& val) {
+    if (val != prevVal) {
+      SetValue(val);
+      prevVal = val;
+    }
+  }
+
+ private:
+  nt::NetworkTableEntry ntEntry;
+  T prevVal;
+
+  void SetValue(const bool& val) { ntEntry.SetBoolean(val); }
+  void SetValue(const double& val) { ntEntry.SetDouble(val); }
```

Comment:
Could just be `bool` and `double` instead of requiring references.

Reply:
Yep, correct, that was a copy-paste error from the string version. Fixed